Portal/Controllers/DropzoneController.cs

```diff
@@ -0,0 +1,101 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+
+namespace Portal.Controllers
+{
+    public class DropzoneController : Controller
+    {
+        private readonly ClaimsPrincipal currentUser;
+
+        public DropzoneController(ClaimsPrincipal currentUser)
+        {
+            this.currentUser = currentUser;
+        }
+        
+        [HttpPost]
+        [Route("[controller]/{customer}/{space}/[action]")]
+        public async Task<IActionResult> Local(int customer, int space, List<IFormFile> file)
+        {
+            if (currentUser.GetCustomerId() != customer)
+            {
+                throw new InvalidOperationException("Customer ID mismatch");
+            }
+            string? errorMessage = null; 
+            string fileNameForSaving = "";
+            try
+            {
+                foreach (var formFile in file)
+                {
+                    if (formFile.Length > 0)
+                    {
+                        //Save file content goes here
+                        fileNameForSaving = Path.GetFileName(formFile.FileName) ?? formFile.FileName;
+                        fileNameForSaving = fileNameForSaving.Replace(' ', '_'); 
+                        
+                        var target = GetTargetFilePath(customer, space, fileNameForSaving);
+                        await using var stream = System.IO.File.Create(target);
+                        await formFile.CopyToAsync(stream);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+            }
+
+            if (errorMessage == null)
+            {
+                return Json(new { message = "completed: " + fileNameForSaving });
+            }
+            return Json(new { error = errorMessage, message = "error: " + fileNameForSaving + ", " + errorMessage });
+            
+        }
+
+        private static string GetTargetFilePath(int customer, int space, string fileNameForSaving)
+        {
+            var targetDir = Path.Combine(Path.GetTempPath(), customer.ToString(), space.ToString());
+            Directory.CreateDirectory(targetDir);
+            var target = Path.Combine(targetDir, fileNameForSaving);
+            return target;
+        }
+
+
+        [HttpPost]
+        [Route("[controller]/{customer}/{space}/[action]")]
+        public async Task<IActionResult> External(int customer, int space, [FromBody] ExternalImage image)
```

Comment:
Is it worth omitting `{customer}` slug and always just take from cookie?

Reply:
I wasn't sure about that... it's there for security, obvs - no uploading to other people's spaces - but also to inform when the request isn't constructed properly "I meant cust 2 space 6 and I thought I'd supplied the right cookie..."